Source/Core/Core/HW/SI/SI_DeviceAMBaseboard.cpp

```diff
@@ -0,0 +1,2375 @@
+                AMMediaboard::GetGameType() == FZeroAXMonster)
+            {
+              // Status
+              m_motor_reply[command_offset + 2] = 0;
+              m_motor_reply[command_offset + 3] = 0;
+
+              // Error
+              m_motor_reply[command_offset + 4] = 0;
+
+              switch (serial_command >> 24)
+              {
+              case 0:
+              case 1:  // Set Maximum?
+              case 2:
+                break;
+
+              // 0x00-0x40: left
+              // 0x40-0x80: right
+              case 4:  // Move Steering Wheel
+                // Left
+                if (serial_command & 0x010000)
+                {
+                  m_motor_force_y = -((s16)serial_command & 0xFF00);
+                }
+                else  // Right
+                {
+                  m_motor_force_y = (serial_command - 0x4000) & 0xFF00;
+                }
+
+                m_motor_force_y *= 2;
+
+                // FFB
+                if (m_motor_init == 2)
+                {
+                  if (serial_interface.GetDeviceType(1) == SerialInterface::SIDEVICE_GC_STEERING)
+                  {
+                    const GCPadStatus pad_status = Pad::GetStatus(1);
+                    if (pad_status.isConnected)
+                    {
+                      const ControlState mapped_strength = (double)(m_motor_force_y >> 8) / 127.f;
+
+                      Pad::Rumble(1, mapped_strength);
+                      INFO_LOG_FMT(SERIALINTERFACE_AMBB,
+                                   "GC-AM: Command 0x31 (MOTOR) mapped_strength:{}",
+                                   mapped_strength);
+                    }
+                  }
+                }
+                break;
+              case 6:  // nice
+              case 9:
+              default:
+                break;
+              // Switch back to normal controls
+              case 7:
+                m_motor_init = 2;
+                break;
+              // Reset
+              case 0x7F:
+                m_motor_init = 1;
+                memset(m_motor_reply, 0, sizeof(m_motor_reply));
+                break;
+              }
+
+              // Checksum
+              m_motor_reply[command_offset + 5] = m_motor_reply[command_offset + 2] ^
+                                                  m_motor_reply[command_offset + 3] ^
+                                                  m_motor_reply[command_offset + 4];
+            }
+          }
+
+          if (length == 0)
+          {
+            data_out[data_offset++] = gcam_command;
+            data_out[data_offset++] = 0x00;
+          }
+          else
+          {
+            if (m_motor_init)
+            {
+              // Motor
+              m_motor_reply[0] = gcam_command;
+              m_motor_reply[1] = length;  // Same out as in size
+
+              memcpy(data_out.data() + data_offset, m_motor_reply, m_motor_reply[1] + 2);
```

Comment:
The size argument might be too large for `m_motor_reply`. (Also, this is another unchecked `data_offset`.)